### ISSUE TITLE
magento2 returns 500 on both triggers

### ISSUE THREAD

**Issue by khanzadyan** [elasticio/magento2-component#31]
To replicate the problem follow the steps below:

1. Take the standard Magento2 component from the production. Create credentials using the test instance username/password (from bitwarden). Verification process goes ok but we know well that it is just a superficial process.

2.  Use any of the two triggers `Get Customers` or `Get Products`. Not important at this stage. Press retrieve a sample.

```
Starting your Docker container ...
Starting slug download ...
Not Available
[INFO  tini (1)] Spawned child process '/run.sh' with pid '7'
Starting slug download ...
Successfully downloaded and extracted slug file
node: /usr/lib/libstdc++.so.6: no version information available (required by node)
node: /usr/lib/libstdc++.so.6: no version information available (required by node)
node: /usr/lib/libstdc++.so.6: no version information available (required by node)
node: /usr/lib/libstdc++.so.6: no version information available (required by node)
node: /usr/lib/libstdc++.so.6: no version information available (required by node)
node: /usr/lib/libstdc++.so.6: no version information available (required by node)

... skipped

start trigger
snapshot  {}
data: undefined },
response:
statusText: 'Internal Server Error',
server: 'Apache',
'transfer-encoding': 'chunked',
{ Error: Request failed with status code 500
at endReadableNT (_stream_readable.js:1047:12)
at process._tickCallback (internal/process/next_tick.js:161:9)
{ adapter: [Function: httpAdapter],
xsrfHeaderName: 'X-XSRF-TOKEN',
headers:
at emitNone (events.js:110:20)
config:
timeout: 0,
maxContentLength: -1,
url: 'http://www.eiotesting.a2hosted.com/magento_2_2_6/rest/V1/customerGroups/search?searchCriteria',
at _combinedTickCallback (internal/process/next_tick.js:102:11)
'User-Agent': 'axios/0.15.3' },
method: 'get',
{ status: 500,
headers:
'content-type': 'application/json; charset=UTF-8' },
config:
at createError (/home/apprunner/node_modules/axios/lib/core/createError.js:15:15)
'x-powered-by': 'PHP/7.0.32',
{ date: 'Tue, 04 Dec 2018 10:15:11 GMT',
at IncomingMessage.emit (events.js:207:7)
{ Accept: 'application/json, text/plain, */*',
at settle (/home/apprunner/node_modules/axios/lib/core/settle.js:18:12)
transformRequest: { '0': [Function: transformRequest] },
validateStatus: [Function: validateStatus],
baseURL: 'http://www.eiotesting.a2hosted.com/magento_2_2_6/rest/V1',
connection: 'close',
at IncomingMessage.handleStreamEnd (/home/apprunner/node_modules/axios/lib/adapters/http.js:186:11)
transformResponse: { '0': [Function: transformResponse] },
xsrfCookieName: 'XSRF-TOKEN',
Authorization: 'Bearer 5jcb645z7u55hi6ey1f5gf9so3vkq5vh',
timeout: 0,
xsrfCookieName: 'XSRF-TOKEN',
method: 'get',
domain: null,
_maxListeners: undefined,
_options: [Object],
_redirectCount: 0,
data: { messages: [Object] } } }
{ adapter: [Function: httpAdapter],
maxContentLength: -1,
headers: [Object],
baseURL: 'http://www.eiotesting.a2hosted.com/magento_2_2_6/rest/V1',
_currentRequest: [Object],
_currentUrl: 'http://www.eiotesting.a2hosted.com' },
transformRequest: [Object],
xsrfHeaderName: 'X-XSRF-TOKEN',
data: undefined },
_eventsCount: 2,
validateStatus: [Function: validateStatus],
request:
Writable {
_events: [Object],
_onNativeResponse: [Function],
transformResponse: [Object],
url: 'http://www.eiotesting.a2hosted.com/magento_2_2_6/rest/V1/customerGroups/search?searchCriteria',
writable: true,
_writableState: [Object],

... skipped

```
This particular error is when `Get Customers` is used. Error is similar in case of the other trigger. The main library `axios` is reporting problems.

**Comment by kirill-levitskiy-elasticio** [comment]
The problem was described in the issue https://github.com/magento/magento2/issues/16910

Endpoints like `customerGroup/search` and `products` need `searchCriteria` to get a full list of items.
For Magento 2.1.X `adding V1/customerGroups/search/?searchCriteria` to get a list is working properly,
but since 2.2.4 this doesn't work anymore and were found different solutions to the problem.
One of them is sending invalid `searchCriteria` like `V1/customerGroups/search/?searchCriteria=searchCriteria` OR a question mark `V1/customerGroups/search/?searchCriteria=?` OR whatever.
The API doesn't care and gives you a full list unless the searchCriteria is VALID.

We need to find correct and flexible solution for different versions of Magento 2.x.

**Comment by khanzadyan** [comment]
I got a feedback from the customer; he needs a working component latest on 12.12. @zubairov and @juhrmacher can we consider this information when scheduling this task for the Sprint run?

**Comment by zubairov** [comment]
@kirill-levitskiy-elasticio if I understand correctly our current version of Magento component does not work with specific Magento versions right?

@khanzadyan who is the customer, can he/she downgrade/upgrade his/her Magento?

**Comment by kirill-levitskiy-elasticio** [comment]
@zubairov correct, triggers in our current version of the Magento component don't work properly due to changes in API at Magento 2.2.4+

**Comment by zubairov** [comment]
thanks @kirill-levitskiy-elasticio , which version of magento do we need them to work now?

**Comment by kirill-levitskiy-elasticio** [comment]
@zubairov I guess, there should be installed Magento older than 2.2.4, but we should check it somehow to be sure. I can check it with available test Magento instances.

**Comment by zubairov** [comment]
thanks, that should be enough

@khanzadyan please speak to customer if he/she only needs it for demo then he/she should downgrade, if not we will decide.

**Comment by khanzadyan** [comment]
@zubairov OK, can we state until which version is our current Magento2 component working? @kirill-levitskiy-elasticio? 

@zubairov This repo is open, so I would refrain from naming any customer by name.

**Comment by khanzadyan** [comment]
@zubairov and @kirill-levitskiy-elasticio we have a second customer with the same problem. His Magento version is 2.2.6. So if you are counting the votes please add one more.

**Comment by kirill-levitskiy-elasticio** [comment]
@khanzadyan as mentioned above, this issue arises with Magento version 2.2.5+.
As temporary workaround:
* Downgrade Magento to 2.2.4;
* Use the component from branch [fix-issue#31](https://github.com/elasticio/magento2-component/tree/fix-issue%2331) with hotfix: c5c9d43

**Comment by khanzadyan** [comment]
@kirill-levitskiy-elasticio I understand it would be possible to do this. However:

* It is not viable to tell the client to downgrade their live Magento 2 install. Too many things can go wrong
* If we give a customised component, then the customer would think we should develop further customisation for them.

@zubairov and @kirill-levitskiy-elasticio we want to avoid the same kind of problem we had previously.
